mlflow/utils/conda.py

```diff
@@ -133,10 +181,29 @@ def get_or_create_conda_env(conda_env_path, env_id=None, capture_output=False):
             )
         )
 
-    env_names = _list_conda_environments()
+    conda_extra_envs = _get_conda_extra_envs(env_root_dir)
+
     project_env_name = _get_conda_env_name(conda_env_path, env_id)
-    if project_env_name not in env_names:
-        _logger.info("=== Creating conda environment %s ===", project_env_name)
+    if env_root_dir is not None:
+        # Append a suffix "-isolated" because if a conda env name exist in
+        # default conda env root dir, then if we set new "CONDA_ENVS_PATH" and run "conda env create"
+        # with the same conda env name, "CondaValueError: prefix already exists" error will
+        # be raised.
+        project_env_name = project_env_name + "-isolated"
```

Comment:
What would happen if an environment named `project_env_name + "-isolated"` exists in the default conda env root?

Reply:
Instead of appending `isolated`, can we hash the environment root directory and append it for uniqueness?

Reply:
> Instead of appending isolated, can we hash the environment root directory path and append it for uniqueness?

Good idea